test/parallel/test-domain-abort-when-disposed.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const common = require('../common');
+const JSStream = process.binding('js_stream').JSStream;
+const assert = require('assert');
+const Socket = require('net').Socket;
+const domain = require('domain');
+
+const handle = new JSStream();
+handle.domain = domain.create();
+handle.domain.dispose();
+
+handle.close = () => {};
+handle.isAlive = () => { throw new Error(); };
+
+const s = new Socket({ handle });
+
+// When AsyncWrap::MakeCallback() returned an empty handle the
+// MaybeLocal::ToLocalChecked() call caused the process to abort. By returning
+// v8::Undefined() it allows the error to propagate to the 'error' event.
+s.on('error', common.mustCall((e) => {
```

Comment:
@refack implementation details of how the `JSStream` instance dictates how the `Socket` connects. Because the `JSStream` instance isn't connected to anything the socket reports `EINVAL`.